crypto/err/err.c

```diff
@@ -224,24 +224,33 @@ static void build_SYS_str_reasons(void)
         str->error = ERR_PACK(ERR_LIB_SYS, 0, i);
         if (str->string == NULL) {
             if (openssl_strerror_r(i, cur, sizeof(strerror_pool) - cnt)) {
```

Comment:
I'd recommend avoid calling openssl_strerror_r with buflen==0.
I am however not sure if the fix should be in openssl_strerror_r instead, just return 0 if buflen==0.